src/app/src/main/kotlin/org/vaccineimpact/api/app/repositories/jooq/JooqModellingGroupRepository.kt

```diff
@@ -142,9 +141,59 @@ class JooqModellingGroupRepository(
         ), scenarioAndData.tableData)
     }
 
-    private fun convertScenarioToResponsibility(scenario: Scenario): Responsibility
+    private fun convertScenarioToResponsibility(scenario: Scenario, responsibilityId: Int): Responsibility
     {
-        return Responsibility(scenario, ResponsibilityStatus.EMPTY, emptyList(), null)
+        val burdenEstimate = getCurrentBurdenEstimate(responsibilityId)
+        val problems: MutableList<String> = mutableListOf()
+        val status = if (burdenEstimate == null)
+        {
+            ResponsibilityStatus.EMPTY
+        }
+        else
+        {
+            if (burdenEstimate.problems.any())
+            {
+                problems.addAll(burdenEstimate.problems)
+                ResponsibilityStatus.INVALID
+            }
+            else
+            {
+                ResponsibilityStatus.VALID
+            }
+        }
+        return Responsibility(scenario, status, problems, burdenEstimate)
+    }
+
+    private fun getCurrentBurdenEstimate(responsibilityId: Int): BurdenEstimateSet?
+    {
+        val records = dsl.select(BURDEN_ESTIMATE_SET_PROBLEM.PROBLEM)
+                .select(BURDEN_ESTIMATE_SET.UPLOADED_ON, BURDEN_ESTIMATE_SET.ID)
+                .from(BURDEN_ESTIMATE_SET)
+                .leftJoin(BURDEN_ESTIMATE_SET_PROBLEM)
+                .on(BURDEN_ESTIMATE_SET_PROBLEM.BURDEN_ESTIMATE_SET.eq(BURDEN_ESTIMATE_SET.ID))
+                .join(RESPONSIBILITY)
+                .on(RESPONSIBILITY.CURRENT_BURDEN_ESTIMATE_SET.eq(BURDEN_ESTIMATE_SET.ID))
+                .where(RESPONSIBILITY.ID.eq(responsibilityId))
+                .fetch()
```

Comment:
I *think* that what you're doing here is complicated enough you can't use the `joinPath` helpers I wrote, but just to check: Are you aware of the `joinPath` helpers and are you comfortable using them? If not, I can give you a quick briefing on them.

Reply:
Have changed to use the helper for the first join, the second one I think needs specifying precisely as there are 2 possible join paths from `responsibility` to `burden_estimate_set`